core/jazz_environment-event-handler/components/safe.js

```diff
@@ -0,0 +1,291 @@
+// =========================================================================
+// Copyright © 2017 T-Mobile USA, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =========================================================================
+"use strict";
+
+const request = require("request");
+const logger = require("./logger.js");
+
+function addSafe(environmentApiPayload, serviceDetails, configData, authToken) {
+  try{
+  return new Promise((resolve, reject) => {
+    if (!configData.TVAULT || !configData.TVAULT.IS_ENABLED) {
+      return resolve({
+        "error": "T-vault is not enabled",
+      });
+    }
+    safeExportable.createSafe(environmentApiPayload, serviceDetails.id, configData, authToken)
+        .then((safeName) => { return safeExportable.getAdmins(environmentApiPayload, serviceDetails.id, configData, authToken, safeName)})
+        .then((result) => { return safeExportable.addAdminsToSafe(environmentApiPayload, configData, authToken, result)})
+        .then((result) => { return resolve(result); })
+        .catch((err) => {
+          logger.error("add safe details failed: " + err);
+          return reject(err);
+        })
+    });
+  } catch (err) {
+    logger.error("add safe details failed: " + err);
+  }
+}
+
+function removeSafe(environmentApiPayload, configData, authToken) {
+  try{
+  return new Promise((resolve, reject) => {
+    if (!configData.TVAULT || !configData.TVAULT.IS_ENABLED) {
+      return resolve({
+        "error": "T-vault is not enabled",
+      });
+    }
+    safeExportable.getEnvDetails(environmentApiPayload, configData, authToken)
+        .then((result) => { return safeExportable.deleteSafe(environmentApiPayload, configData, authToken, result)})
+        .then((result) => { return resolve(result); })
+        .catch((err) => {
+          logger.error("removing safe details failed: " + err);
+          return reject(err);
+        })
+    });
+  } catch (err) {
+    logger.error("removing safe details failed: " + err);
+  }
+}
+
+function createSafe(environmentPayload, service_id, configData, authToken) {
+  return new Promise((resolve, reject) => {
+    var updatePayload = {};
+    const safeName = environmentPayload.service + '_' + environmentPayload.domain;
```

Comment:
The safe name over here will be same for all the environments. I believe we need a environment specific safename.

Reply:
As per discussion, we decided to remove env from a safe name too as per my understanding. I guess safe name differentiation is not needed, as GET Safe API is env specific.

Reply:
@dmpl1 @SiniWilson We would need a safe per environment. Also see story description, environmentId also needs to be included in safe name.

Reply:
Addressed